### PR TITLE
Add CPython dict tests

### PR DESCRIPTION
**Pull request by S-YOU**
- This required decorators, need to rebase once decorators are ok.
- Currently unsupported tests are marked as expectedFailure.
- Minor changes on unittest and contextlib to make this work with least modifications.
- Haven't included this test in Makefile yet.

```
test_bool (__main__.DictTest) ... ok
test_clear (__main__.DictTest) ... ok
test_constructor (__main__.DictTest) ... ok
test_contains (__main__.DictTest) ... ok
test_copy (__main__.DictTest) ... expected failure
test_empty_presized_dict_in_freelist (__main__.DictTest) ... ok
test_fromkeys (__main__.DictTest) ... expected failure
test_get (__main__.DictTest) ... ok
test_getitem (__main__.DictTest) ... ok
test_has_key (__main__.DictTest) ... expected failure
test_items (__main__.DictTest) ... ok
test_keys (__main__.DictTest) ... expected failure
test_le (__main__.DictTest) ... expected failure
test_len (__main__.DictTest) ... ok
test_literal_constructor (__main__.DictTest) ... expected failure
test_missing (__main__.DictTest) ... expected failure
test_mutatingiteration (__main__.DictTest) ... ok
test_pop (__main__.DictTest) ... expected failure
test_popitem (__main__.DictTest) ... expected failure
test_repr (__main__.DictTest) ... ok
test_resize1 (__main__.DictTest) ... ok
test_resize2 (__main__.DictTest) ... ok
test_setdefault (__main__.DictTest) ... expected failure
test_setdefault_atomic (__main__.DictTest) ... expected failure
test_track_dynamic (__main__.DictTest) ... ok
test_track_literals (__main__.DictTest) ... ok
test_track_subtypes (__main__.DictTest) ... ok
test_tuple_keyerror (__main__.DictTest) ... expected failure
test_update (__main__.DictTest) ... expected failure
test_values (__main__.DictTest) ... expected failure

----------------------------------------------------------------------
Ran 30 tests in 0.011339s

OK (expected failures=14)
```

### CODE REVIEW

File: third_party/stdlib/test/test_dict.py [hunk 1]
```diff
@@ -0,0 +1,734 @@
...
+            GeneralMappingTests,
+            SubclassMappingTests,
+        )
+
+if __name__ == "__main__":
+    import traceback
+    try:
+        test_main()
+    except:
+        traceback.print_exc()
```
Comment (S-YOU) [inline third_party/stdlib/test/test_dict.py]: Note to self: Remove this try block after rebase.

**Comment by S-YOU** [comment]
Could you please explain me a bit what is the "converge on files" on this context mean?

**Comment by trotterdylan** [comment]
I'd like these files to eventually be exact copies of the ones from CPython/PyPy. So that implies we should work toward undoing the diffs we created to get them working when we originally committed them. In this sense the original files and the ones in our third_party/ subdirs should converge over time.

Does that make sense?

**Comment by S-YOU** [comment]
Got it. We might need to create dummy gc package to be able to do it, I. guess

**Pull request merged**